docs/geps/gep-03.rst

```diff
@@ -0,0 +1,342 @@
+``description``, and the ``YYYY-MM-DD``-formatted date on which it first took effect.
+Values usually change over time; each time a value is changed, another ``YYYY-MM-DD``
+entry is added.
+
+1. The ``name`` key has two sub-keys ``de`` and ``en``, which are
+
+   - short names without re-stating the realm of the parameter (e.g.
+     "Arbeitslosengeld II" or "Kinderzuschlag");
+   - not sentences;
+   - correctly capitalised.
+
+   Example (from ``arbeitsl_geld_2``):
+
+   .. code-block:: yaml
+
+      eink_anr_frei:
+        name:
+          de: Anrechnungsfreie Einkommensanteile
+          en: Income shares not subject to transfer withdrawal
+
+2. The ``description`` key has two sub-keys ``de`` and ``en``, which
+
+   - are exhaustive explanations of the parameter;
+   - show the § and Gesetzbuch of that parameter (including its entire history if the
+     location has changed);
+   - mention bigger amendments / Neufassungen and be as helpful as possible to
+     make sense of that parameter.
+
+     Example:
+
+     .. code-block:: yaml
+
+        eink_anr_frei:
+
+          description:
+            de: >-
+              Einkommensanteile, die anrechnungsfrei bleiben. § 30 SGB II. Seit 01.10.2005 zudem
+              definiert durch Freibetrag in § 11 SGB II, siehe auch § 67 SGB II. Seit 01.04.2011
+              § 11b (2) SGB II (neugefasst durch B. v. 13.05.2011 BGBl. I S. 850. Artikel 2
+              G. v. 24.03.2011 BGBl. I S. 453).
+            en: >-
+              Shares of income which do not lead to tapering of Arbeitslosengeld II benefits.
+
+3. The ``unit`` key informs on the unit of the values (Euro or DM if monetary,
+   share of some other value, ...).
+
+   - In rare cases (e.g. child benefit age threshold), it might be omitted.
+   - Some values used at this point: ``Euro``, ``DM``, ``share``, ``hours``,
+     ``square meters``
+
+   Example:
+
+   .. code-block:: yaml
+
+    kindergeld:
+      name:
+        de: Kindergeld, Betrag je nach Reihenfolge der Kinder.
+      unit: Euro
+
+4. The (optional) ``type`` key may contain a reference to a particular function that
+   is implemented. Examples are ``piecewise_linear`` or ``piecewise_quadratic``
+
+5. The (optional) ``reference_period`` key informs on the reference period of the
+   values, if applicable
+
+   Possible values:
+   - ``year``
+   - ``month``
+   - ``week``
+   - ``day``
+
+   Example:
+
+   .. code-block:: yaml
+
+    kindergeld_stundengrenze:
+      name:
+        de: Wochenstundengrenze für Kindergeldanspruch
+        [...]
+      reference_period: week
+
+6. The (optional) ``access_prior_parameters`` can be used to make the parameter of a
+   previous point in time (relative to the date specified available in
+   ``set_up_policy_environment``) available within GETTSIM functions. It requires the
+   ``reference_period`` (one of ``y``, ``m``, ``w``, ``d``) and the ``number_of_lags``.
```

Comment:
And another thought: Should those be written out: ``year``, ``month``, etc.

Reply:
I'd vote no because we would have to start thinking about singular / plural then...